cpp/src/arrow/flight/transport/grpc/grpc_server.cc

```diff
@@ -244,12 +244,12 @@ class GrpcServiceHandler final : public FlightService::Service {
     // Write flight info to stream until listing is exhausted
     while (true) {
       ProtoType pb_value;
-      std::unique_ptr<UserType> value;
-      GRPC_RETURN_NOT_OK(iterator->Next(&value));
-      if (!value) {
+      arrow::Result<std::unique_ptr<UserType>> value = iterator->Next();
+      GRPC_RETURN_NOT_OK(value.status());
```

Comment:
In general I'd rather we not declare any variables of type `Result<>` and instead stick closer to the prior style, e.g. `GRPC_RETURN_NOT_OK(iterator->Next().Value(&value));`